buf.gen.yaml

```diff
@@ -1,18 +1,18 @@
 ---
 version: "v1"
 plugins:
-  - remote: "buf.build/library/plugins/go:v1.27.1-1"
+  - plugin: "buf.build/protocolbuffers/go:v1.30.0"
```

Comment:
Newer versions of the `buf` CLI use the `plugin` key for both local and remote plugin references.

This PR also updates the plugins to the Buf-managed plugins and also updates them to the latest version:

- https://buf.build/protocolbuffers/go
- https://buf.build/grpc/go
- https://buf.build/bufbuild/validate-go
- https://buf.build/grpc-ecosystem/gateway
- https://buf.build/grpc-ecosystem/openapiv2